src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -144,25 +160,26 @@ private Future<Void> processInventory(CompositePoLine compPOL, boolean deleteHol
         })
         .mapEmpty();
     }
-    if (PoLineCommonUtil.isItemsUpdateRequired(compPOL)) {
-      if (compPOL.getCheckinItems()) { // independent workflow
-        return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
-      } else { // synchronized workflow
-        return (deleteHoldings) ? processInventoryHoldingWithItems(compPOL, requestContext) : processInventoryOnlyWithItems(compPOL, requestContext);
-      }
+    boolean itemsUpdateRequired = PoLineCommonUtil.isItemsUpdateRequired(compPOL);
+    boolean holdingsUpdateRequired = PoLineCommonUtil.isHoldingsUpdateRequired(compPOL);
+    // itemsUpdateRequired => holdingsUpdateRequired
+    if (!itemsUpdateRequired && !holdingsUpdateRequired) {
+      return Future.succeededFuture();
     }
-    if (PoLineCommonUtil.isHoldingsUpdateRequired(compPOL)) {
```

Comment:
we need to check holdingsUpdateRequired here, correct?

Reply:
seems previous version of this method is more readable

Reply:
Agree. I'll try to farther improve or rollback to previous version.

Reply:
I see you back to previous implementation, resolvign comment